redoctober_test.go

```diff
@@ -0,0 +1,701 @@
+// Tests the Red October API. These tests ensure consistency in the API's
+// return status codes and data. They do not explicitly test for correctness
+// of the functions called (that is handled by the tests in /core). Running
+// these tests first require that you build this project and have a recent
+// binary of it in either the /redoctober folder or in $GOPATH/bin/redoctober.
+
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/cloudflare/redoctober/core"
+)
+
+const baseURL = "https://localhost:8080/"
+
+var (
+	// Inputs that are used in multiple test cases are put here to reduce verbosity.
```

Comment:
I suspect this test file could be made easier to read and understand by moving to a table-driven test model (https://github.com/golang/go/wiki/TableDrivenTests). You'd have a struct representing each test case, and a single loop using your current functions to execute a test over the struct.